Tracking/include/Tracking/Reco/LinearSeedFinder.h

```diff
@@ -0,0 +1,144 @@
+#pragma once
+
+//---< Framework >---//
+#include "Framework/Configure/Parameters.h"
+#include "Framework/Event.h"
+#include "Framework/EventProcessor.h"
+
+//---< SimCore >---//
+#include "SimCore/Event/SimTrackerHit.h"
+
+//--- LDMX ---//
+#include "Tracking/Reco/TrackingGeometryUser.h"
+
+//---< STD C++ >---//
+#include <iostream>
+
+//--- LDMX ---//
+#include "Ecal/Event/EcalHit.h"
+#include "TFile.h"
+#include "TTree.h"
+#include "Tracking/Event/Measurement.h"
+#include "Tracking/Event/StraightTrack.h"
+#include "Tracking/Reco/TrackingGeometryUser.h"
+#include "Tracking/Reco/TruthMatchingTool.h"
+
+namespace tracking {
+namespace reco {
+
+class LinearSeedFinder : public TrackingGeometryUser {
+ public:
+  /**
+   * Constructor.
+   *
+   * @param name The name of the instance of this object.
+   * @param process The process running this producer.
+   */
+  LinearSeedFinder(const std::string &name, framework::Process &process);
+
+  /// Destructor
+  ~LinearSeedFinder();
+
+  /**
+   *
+   */
+  void onProcessStart() override;
+
+  /**
+   *
+   */
+  void onProcessEnd() override;
+
+  /**
+   * Configure the processor using the given user specified parameters.
+   *
+   * @param parameters Set of parameters used to configure this processor.
+   */
+  void configure(framework::config::Parameters &parameters) override;
+
+  /**
+   * Run the processor and create a collection of results which
+   * indicate if a charge particle can be found by the recoil tracker.
+   *
+   * @param event The event to process.
+   */
+  void produce(framework::Event &event) override;
+
+ protected:
+  // Function to find seeds based on 2 Recoil points and 1 EcalRecHit
+  ldmx::StraightTrack SeedTracker(
+      const std::tuple<std::array<double, 3>, ldmx::Measurement,
```

Comment:
can we make that "3" a variable? If in some lucky case we get to have another tracking layer?

Reply:
It's ok let's keep this as is